lib/danger-packwerk/private.rb

```diff
@@ -1,13 +1,27 @@
-# typed: strong
+# typed: strict
 
 require 'danger-packwerk/private/deprecated_references'
 require 'danger-packwerk/private/default_offenses_formatter'
+require 'constant_resolver'
 
 module DangerPackwerk
   #
   # Anything within the Private module is subject to change.
   #
   module Private
+    extend T::Sig
+
+    sig { returns(ConstantResolver) }
+    def self.constant_resolver
+      @constant_resolver = T.let(@constant_resolver, T.nilable(ConstantResolver))
+      @constant_resolver ||= begin
+        load_paths = Packwerk::ApplicationLoadPaths.extract_relevant_paths(Dir.pwd, 'test')
+        ConstantResolver.new(
```

Comment:
Originally I wasn't planning on pulling on `ConstantResolver` directly and was going to have the client inject it.

This ends up being easier, especially considering the client will already have this dependency and its implications via packwerk itself.